docs/proposals/logdir.md

```diff
@@ -0,0 +1,163 @@
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/logdir.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# LogDir
+
+## Abstract
+
+A proposal for implementing `LogDir`. A `LogDir` can preserve and manage a pod's log files.
+
+Several existing issues and PRs were already created regarding that particular subject:
+* Collect logfiles inside the container [#12567](https://github.com/kubernetes/kubernetes/issues/12567)
+* Add LogDir [#13010](https://github.com/kubernetes/kubernetes/pull/13010#issuecomment-133997188)
+
+Author: WuLonghui (@wulonghui)
+
+## Motivation
+
+Some applications will write log to files, so need to add  `LogDir`. It can add some useful infomation to make logging agent easy to collect the logs, also we can manage the log files(like log rotation and maximum log file sizes).
+
+## Implementation
+
+Create `LogDir` as a policy  to `api.VolumeMount` struct :
+
+```
+// VolumeMount describes a mounting of a Volume within a container.
+type VolumeMount struct {
+  // Required: This must match the Name of a Volume [above].
+  Name string `json:"name"`
+  // Optional: Defaults to false (read-write).
+  ReadOnly bool `json:"readOnly,omitempty"`
+  // Required.
+  MountPath string `json:"mountPath"`
+  // Optional: Policies of VolumeMount.
+  Policy *VolumeMountPolicy `json:"policy,omitempty"`
+}
+
+// VolumeMountPolicy describes policies of a VolumeMount.
+type VolumeMountPolicy struct {
+  // Optional: LogDir policy.
+  LogDir *LogDirPolicy `json:"logDir,omitempty"`
+}
+
+// LogDirPolicy describes a policy of logDir, include log rotation and maximum log file size.
+type LogDirPolicy struct {
+  // Optional: Glob pattern of log files.
+  Glob string `json:"glob,omitempty"`
+  // Optional: Log rotation.
+  Rotate string `json:"rotate,omitempty"`
+  // Optional: Maximum log file size.
+  MaxFileSize int `json:"maxFileSize,omitempty"`
+}
+
+```
+
+If users set the LogDir policy of the VolumeMount in a container:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-app
+spec:
+  containers:
+  - name: container1
+    image: ubuntu:14.04
+    command: ["bash", "-c", "i=\"0\"; while true; do echo \"`hostname`_container1: $i \"; date --rfc-3339 ns >> /varlog/container1.log; sleep 4; i=$[$i+1]; done"]
+    volumeMounts:
+    - name: log-storage
+      mountPath: /varlog
+      policy:
+        logDir: {}
+    securityContext:
+        privileged: true
+  volumes:
+  - name: log-storage
+    emptyDir: {}
+```
+
+Kubelet will create a symbolic link to the volume path in `/var/log/containers`.
+
+```
+/var/log/containers/<pod_full_name>_<contianer_name> => <volume_path>

```

Comment:
Also can this be deeper structure or does that defeat things like fluentd?  e.g. /var/lib/kubelet/logdirs/pod-id/container-name